test/EndaomentAdmin.test.js

```diff
@@ -0,0 +1,265 @@
+      newOwner,
+      { from: admin }
+    );
+    expectEvent(transferReceipt, "TransferInitiated", { newOwner });
+  });
+
+  it("denies non-Owner to intiate ownership transfer", async function() {
+    await expectRevert.unspecified(
+      this.endaomentAdmin.transferOwnership(newOwner, { from: newOwner })
+    );
+  });
+
+  it("prevents non-Owner from cancelling ownership transfer", async function() {
+    await expectRevert.unspecified(
+      this.endaomentAdmin.cancelOwnershipTransfer({ from: newOwner })
+    );
+  });
+
+  it("allows Owner to cancel ownership transfer", async function() {
+    await this.endaomentAdmin.transferOwnership(newOwner, { from: admin });
+
+    const cancelReceipt = await this.endaomentAdmin.cancelOwnershipTransfer({
+      from: admin,
+    });
+
+    expectEvent(cancelReceipt, "TransferCancelled", {
+      _newPotentialOwner: newOwner,
+    });
+  });
+
+  it("prevents non-Owner from cancelling ownership transfer", async function() {
+    await this.endaomentAdmin.transferOwnership(newOwner, { from: admin });
+
+    await expectRevert.unspecified(
+      this.endaomentAdmin.cancelOwnershipTransfer({ from: newOwner })
+    );
+  });
+
+  it("denies non-newOwner to accept ownership transfer", async function() {
+    await this.endaomentAdmin.transferOwnership(newOwner, { from: admin });
+
+    await expectRevert.unspecified(
+      this.endaomentAdmin.acceptOwnership({ from: admin })
+    );
+  });
+
+  it("allows only newOwner to accept ownership transfer, and updates Owner successfully", async function() {
+    await this.endaomentAdmin.transferOwnership(newOwner, { from: admin });
+
+    await expectRevert.unspecified(
+      this.endaomentAdmin.acceptOwnership({ from: admin })
+    );
+
+    const ownershipTransferReciept = await this.endaomentAdmin.acceptOwnership({
+      from: newOwner,
+    });
+    expectEvent(ownershipTransferReciept, "OwnershipTransferred", {
+      previousOwner: admin,
+      newOwner,
+    });
+    const updatedOwner = await this.endaomentAdmin.getOwner({ from: newOwner });
+
+    assert.equal(newOwner, updatedOwner);
+  });
+
+  it("denies non-Owner from setting a role", async function() {
+    await expectRevert.unspecified(
+      this.endaomentAdmin.setRole(0, admin, { from: newOwner })
+    );
+  });
+
+  it("allows Owner to set Role.ADMIN, allows retrieval of ADMIN address", async function() {
+    await this.endaomentAdmin.setRole(0, admin, { from: admin });
+    const adminRole = await this.endaomentAdmin.getAdmin({ from: admin });
+
+    assert.equal(adminRole, admin);
+  });
+
+  it("allows Owner to set Role.PAUSER, allows retrieval of PAUSER address", async function() {
+    await this.endaomentAdmin.setRole(1, pauser, { from: admin });
+    const pauserRole = await this.endaomentAdmin.getPauser({ from: admin });
+
+    assert.equal(pauserRole, pauser);
+  });
+
+  it("allows Owner to set Role.ACCOUNTANT, allows retrieval of ACCOUNTANT address", async function() {
```

Comment:
replace with a single check of getRoleAddress for each of these with a test of a single role.